### DISCUSSION TITLE
Isn't there `useBalances` like `useContractReads`?

### DISCUSSION THREAD

**Discussion by madiha-right** [wevm/wagmi#1266] [Q&A] [answered]
Hi I'm trying to get ETH, WETH balance at the same time from a single address using `useBalance` hook and I don't think there is any hooks in wagmi that is able to fetch both ETH, WETH parallelly. Isn't there any hooks or options like `useContractReads` for `useBalance`?

**Comment by tmm** [comment, marked as answer]
There isn't a `useBalances` right now. I'd recommend fetching the native currency balance (e.g. ether) with `useBalance` and any other balances with `useBalance` (for a single additional balance) or `useContractReads` (for many additional balances).

**Reply by madiha-right** [reply]
@tmm thanks for the reply! Hope to see `useBalances` hook soon enough!

**Reply by ryanberckmans** [reply]
@tmm a `useBalances` hook would be great to fetch the native currency balance and many additional token balances in a single RPC call.

Additionally, it appears that while `useBalance({ watch: true })` for a chain with no websocket provider will create a singleton `eth_blockNumber` poll to power all hook instances, a chain with a websocket provider seems to create one `newHeads` subscription per hook instance instead of having a singleton subscription shared among all hook instances with `watch: true`. Anecdotally, I've noticed newHeads subscriptions consuming 70% of all compute units in our alchemy dashboard, granted our app fetches balances for a lot of different tokens.
